lambdas/catalog-updater/__tests__/catalog-updater-test.js

```diff
@@ -0,0 +1,222 @@
+        // suppress console logging
+        jest.spyOn(global.console, 'log').mockImplementation(() => jest.fn())
+
+        // not in catalog/ directory, should all fail
+        expect(index.swaggerFileFilter({Key: 'swagger.json'})).toBe(false)
+        expect(index.swaggerFileFilter({Key: 'swagger.yaml'})).toBe(false)
+        expect(index.swaggerFileFilter({Key: 'swagger.yml'})).toBe(false)
+        expect(index.swaggerFileFilter({Key: 'swagger.json'})).toBe(false)
+        expect(index.swaggerFileFilter({Key: 'not-swagger'})).toBe(false)
+        expect(index.swaggerFileFilter({Key: 'not-swagger.json.misleading'})).toBe(false)
+
+        // in catalog/ directory, only the swagger json/yaml/yml should pass
+        expect(index.swaggerFileFilter({Key: 'catalog/swagger.json'})).toBe(true)
+        expect(index.swaggerFileFilter({Key: 'catalog/swagger.yaml'})).toBe(true)
+        expect(index.swaggerFileFilter({Key: 'catalog/swagger.yml'})).toBe(true)
+        expect(index.swaggerFileFilter({Key: 'catalog/swagger.json/nested-swagger-counts-too.json'})).toBe(true)
+        expect(index.swaggerFileFilter({Key: 'catalog/not-swagger.json/fakeout'})).toBe(false)
+        expect(index.swaggerFileFilter({Key: 'catalog/not-swagger'})).toBe(false)
+        expect(index.swaggerFileFilter({Key: 'catalog/not-swagger.json.misleading'})).toBe(false)
+    });
+})
+
+describe('getSwaggerFile', () => {
+
+    afterEach(() => {
+        index.s3.getObject = jest.fn(() => { throw new Error('getObject called before init') })
+    })
+
+    test('should recognize generic swagger files after fetching them from S3', async () => {
+        // generic swagger is swagger not managed by api gateway; we detect this based on it not having any
+        // x-amazon-apigateway-integration fields in the path
+
+        let fileBody = fs.readFileSync(path.join(__dirname, '/generic-swagger.json')),
+            internalRepr
+
+        index.s3.getObject = jest.fn().mockReturnValueOnce(promiser({ Body: fileBody }))
+
+        internalRepr = await index.getSwaggerFile({Key: 'catalog/swagger.json'})
+
+        expect(internalRepr.id).toBe(index.hash('catalog/swagger.json'))
+        expect(internalRepr.apiStageKey).toBeUndefined()
+        expect(internalRepr.generic).toBe(true)
+        expect(internalRepr.body).toEqual(JSON.parse(fileBody.toString()))
+    })
+
+    test('should recognize custom domain name swagger files after fetching them from S3', async () => {
+        // custom domain name swagger files have their api id and stage name in their filename
+        // in the future, we should vastly prefer this over way of recognizing the file over the other forms
+
+        let fileBody = fs.readFileSync(path.join(__dirname, '/api-gateway-swagger.json')),
+            internalRepr
+
+        index.s3.getObject = jest.fn().mockReturnValueOnce(promiser({ Body: fileBody }))
+
+        internalRepr = await index.getSwaggerFile({Key: 'catalog/a1b2c3d4e5_customdomain.json'})
```

Comment:
Update to `stageName` or something else that indicates that this is in fact a stage and not the custom domain name.